### ISSUE TITLE
loginprompt.php

### ISSUE THREAD

**Issue by HGustavs** [HGustavs/LenaSYS#2446]
Does loginprompt.php really do anything?

Content of file seems useful, but is it used?

Should we put that contents in other file?

**Comment by c14caran** [comment]
Loginprompt is meant to prevent a user from going to a page that needs a logged in person to function.
A rewrite could be done to improve it's functionality.
Maybe some checks to see if the user is logged in or not.
If this is added it could be included in the beginning of all php files that needs the user to be logged in.
It might be good to keep in one file so any files that need this functionality can easily include the file to implement the wanted behaviour.

**Comment by HGustavs** [comment]
My question is In what situation should this be called?

Im thinking that if someone navigates to a page that person is not allowed to see, does the system not show a login box already?

That is what should hapen IMHO. This code redirects to some random page. I prefer just showing the login dialog, not redirecting to another page.

**Comment by c14caran** [comment]
Ett exempel på detta är att navigera till https://dugga.iit.his.se/Dugga/DuggaSys/duggaed.php utan att vara inloggad. Då får man endast en blank sida. Det vore bra att få komma till hemsidan istället.

**Comment by HGustavs** [comment]
Im thinking that the login prompt should be shown, not a redirect.

Redirection is useless if you have access but is not logged in, but useful if you dont have access... we want to help legitimate users over non-legitimate users.

**Comment by HGustavs** [comment]
~~~
	<!-- Login Dialog START -->
	<?php 
		include '../Shared/loginbox.php';
	?>
	<!-- Login Dialog END -->
~~~

as can be seen, in the code above, from duggaed.php the login prompt is loaded already in all pages, but hidden, the only thing that needs to be done is to show the login prompt if the user is not logged in.

Either by running the code for showing the login prompt in php, or by showing the login prompt if no data is returned from the service e.g. duggaedservice.php

...

**Comment by HGustavs** [comment]
![screenshot 2017-03-30 14 19 28](https://cloud.githubusercontent.com/assets/6978801/24503814/fa953390-1553-11e7-8281-58ebca27713b.png)

Ideally the same code should happen as when one presses the red-man-icon.

**Comment by a15tobli** [comment]
Solved in #3252, closing issue.
